### PR TITLE
Fix server cannot exit properly when enabling cluster mode

### LINKED ISSUE #965: The server cannot exit properly when enabling cluster mode
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-kvrocks/issues) and found no similar issues.

### Version

a8e06c7e373f374e1528861fb96942ebbbb63b2f

### Minimal reproduce step

1. Enable the cluster mode
2. Start the migrate thread
3. Exit the server

### What did you expect to see?

No error happens when shut downing the server with cluster mode

### What did you see instead?

As mentioned in comment: As mentioned in https://github.com/apache/incubator-kvrocks/issues/946#issuecomment-1272444901, we can see the migrate thread was aborted after the server exited.

### Anything Else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by git-hulk**
This closes #965

### CODE REVIEW

**Comment by git-hulk** [comment]
Can review commits one by one, this PR also solves some typos and naming issue.

**Review by tisonkun — DISMISSED** [review]

File: src/slot_migrate.h [hunk 1]
```diff
@@ -145,6 +145,13 @@ class SlotMigrate : public Redis::Database {
   };
   ParserState stat_ = ArrayLen;
 
+  enum ThreadState {
```
Comment (caipengbo) [inline src/slot_migrate.h]: `enum class` is better?

Comment (git-hulk) [reply src/slot_migrate.h]: Sure, it makes sense. Please take a look again

**Review by PragmaTwice — APPROVED** [review]

**Review by caipengbo — APPROVED** [review]

**Comment by git-hulk** [comment]
Thanks all, merging...

**Pull request merged**